BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Tracing/DiagnosticsModule/DiagnosticsEventListener.cs

```diff
@@ -81,5 +80,39 @@ protected override void OnEventSourceCreated(EventSource eventSource)
 
             base.OnEventSourceCreated(eventSource);
         }
+
+        /// <summary>
+        /// This method checks if the given EventSource Name matches known EventSources that we want to subscribe to.
+        /// </summary>
+        private static bool ShouldSubscribe(EventSource eventSource)
+        {
+            if (eventSource.Name.StartsWith("Microsoft-A", StringComparison.Ordinal))
```

Comment:
Why do we need this check?

Reply:
we don't NEED it, but it was meant to quickly dismiss any other source names. (rejecting on the 1st if, instead of on the 17th)